components/controls/ListFooter.tsx

```diff
@@ -34,60 +34,61 @@ export interface Props {
   total?: number;
 }
 
-export default function ListFooter({ ready = true, ...props }: Props) {
-  const handleLoadMore = (event: React.SyntheticEvent<HTMLElement>) => {
-    event.preventDefault();
-    event.currentTarget.blur();
-    if (props.loadMore) {
-      props.loadMore();
-    }
-  };
-
-  const handleReload = (event: React.SyntheticEvent<HTMLElement>) => {
-    event.preventDefault();
-    event.currentTarget.blur();
-    if (props.reload) {
-      props.reload();
-    }
-  };
-
-  const hasMore = props.total && props.total > props.count;
-
-  const loadMoreLink = (
-    <a className="spacer-left" data-test="show-more" href="#" onClick={handleLoadMore}>
-      {translate('show_more')}
-    </a>
-  );
-
-  const reloadLink = (
-    <a className="spacer-left" data-test="reload" href="#" onClick={handleReload}>
-      {translate('reload')}
-    </a>
-  );
-
-  const className = classNames(
-    'spacer-top note text-center',
-    { 'new-loading': !ready },
-    props.className
-  );
+export default function ListFooter(props: ListFooterProps) {
+  const { className, count, loading, needReload, total, ready = true } = props;
+  const hasMore = total && total > count;
 
   let link;
-
-  if (props.needReload && props.reload) {
-    link = reloadLink;
+  if (loading) {
+    link = (
+      <span className="spacer-left">
+        {needReload ? translate('reload') : translate('show_more')}
+      </span>
+    );
+  } else if (needReload && props.reload) {
+    link = (
+      <a
+        className="spacer-left"
+        data-test="reload"
+        href="#"
+        onClick={(event: React.SyntheticEvent<HTMLElement>) => {
+          event.preventDefault();
+          event.currentTarget.blur();
+          if (props.reload) {
+            props.reload();
+          }
+        }}>
+        {translate('reload')}
+      </a>
+    );
```

Comment:
Why not use a `ButtonLink`, make it disabled when `loading`?
It would simplify the callbacks significantly, too.